File: 240306/삼각형 만들기/create-triangle.py
n=int(input())

location=[list(map(int,input().split())) for _ in range(n)]

ans=0

def area(p1,p2,p3):
    return abs(p1[0]-p2[0])*p3[1]

def check_x(i,j,k):
    p1,p2,p3=location[i],location[j],location[k]
    if p1[1]==p2[1] or p2[1]==p3[1] or p3[1]==p1[1]:
        return True     
    return 0

def check_y(i,j,k):
    p1,p2,p3=location[i],location[j],location[k]
    if p1[0]==p2[0] or p2[0]==p3[0] or p3[0]==p1[0]:
        return True     
    return False

for i in range(n):
    for j in range(n):
        for k in range(n):
            if i==j or j==k or i==k:
                continue
            if check_x(i,j,k) and check_y(i,j,k):
                ans=max(ans,area(location[i],location[j],location[k]))

if ans==0:
    print(0)
else:
    print(ans)

File: 260118/삼각형 만들기/create-triangle.js
const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split('\n');
const n = Number(input[0]);
const points = input.slice(1, 1 + n).map(line => line.split(' ').map(Number));
// Please Write your code here.

let ans = 0;

for(let i=0; i<n; i++){
    for(let j=i+1; j<n; j++){
        for(let k=j+1; k<n; k++){
            let area;
            const [x1,y1] = points[i], [x2,y2] = points[j], [x3,y3]= points[k];

            if(y1===y2 && (x1===x3|| x2===x3)){
                area = Math.abs((x2-x1)*(y3-y1)/2);
            }
            else if(x1===x2 && (y1===y3|| y2===y3)){
                area = Math.abs((y2-y1)*(x3-x1)/2);
            }else{
                continue;
            }

            ans = Math.max(ans,area);
        }
    }
}

console.log(ans*2);